src/emqx_listeners.erl

```diff
@@ -173,9 +173,15 @@ with_port({Addr, Port}, Opts = #{socket_opts := SocketOption}) ->
 restart() ->
     lists:foreach(fun restart_listener/1, emqx:get_env(listeners, [])).
 
--spec(restart_listener(listener()) -> any()).
+% emqx_listeners:find_by_id(Identifier)
+-spec(restart_listener(listener() | string() | binary()) -> ok | {error, any()}).
 restart_listener(#{proto := Proto, listen_on := ListenOn, opts := Options}) ->
-    restart_listener(Proto, ListenOn, Options).
+    restart_listener(Proto, ListenOn, Options);
+restart_listener(Identifier) ->
+    case emqx_listeners:find_by_id(Identifier) of
+        false    -> {error, {no_such_listener, Identifier}};
+        Listener -> restart_listener(Listener)
+    end.
 
 -spec(restart_listener(esockd:proto(), esockd:listen_on(), [esockd:option()]) -> any()).
```

Comment:
There is no caller which makes use of the Pid returned in `{ok, Pid}` from this function.
Please help to fix this spec to `ok | {error, any()}`.
then add a function like below for line 193 and line 196:
```
ok(ok) -> ok;
ok({ok, _}) -> ok;
ok(Error) -> Error.
```

Reply:
then there should be no need for the `{ok, _} ->` case clause in `emqx_mgmt_cli`